### PR TITLE
[Doc] Put back content from old docs/lang/api/atomic.md

### PR DESCRIPTION
**Pull request by strongoier**
Related issue = #3318

This PR puts back content from old `docs/lang/api/atomic.md` into our current type system section. The format and content organization are refined to fit current doc better. `ti.atomic_max/min` is also added into the doc.

### CODE REVIEW

**Review by ailzhang — APPROVED** [review]
LGTM！
cc: @Vissidarte-Herman but we might want to refine this file as a whole later?

File: docs/lang/articles/basic/type.md [hunk 1]
```diff
@@ -185,6 +185,75 @@ pow(x, y)  # Same as `x ** y`.
...
+| `ti.atomic_or(x, y)`  | atomically compute <code>x &#124; y</code>, store the result in `x`, and return the old value of `x` |
+| `ti.atomic_xor(x, y)` | atomically compute `x ^ y`, store the result in `x`, and return the old value of `x`                 |
+| `ti.atomic_max(x, y)` | atomically compute `max(x, y)`, store the result in `x`, and return the old value of `x`             |
+| `ti.atomic_min(x, y)` | atomically compute `min(x, y)`, store the result in `x`, and return the old value of `x`             |
+
+:::note
+
+Supported atomic operations on each backend:
+
+| type | CPU/CUDA | OpenGL | Metal | C source |
```
Comment (sjwsl) [inline docs/lang/articles/basic/type.md:247]: Add Vulkan, i16, and f16?

**Review by sjwsl — COMMENTED** [review]
> ````

**Comment by strongoier** [comment]
> LGTM！ cc: @Vissidarte-Herman but we might want to refine this file as a whole later?

Ah yes. I'll stop at this point. We just need to remember that we already have something, so there is no need to start from scratch if we decide to write related materials.

File: docs/lang/articles/basic/type.md (hunk 1 above)
Comment (strongoier) [reply docs/lang/articles/basic/type.md:247]: Yep they should be added maybe in a future PR.

**Pull request merged**